arelle/plugin/validate/ROS/rules/ros.py

```diff
@@ -367,14 +371,10 @@ def rule_ros19(
         principal_currency_value_count = unit_counts[principal_currency_value]
         for unit, count in unit_counts.items():
             if count > principal_currency_value_count:
-                modelObjects = principal_currency_facts
-                message = _("'PrincipalCurrencyUsedInBusinessReport' has a value of %(principal_currency_value)s, "
-                            "which must match the functional(majority) unit of the financial statement.")
-                continue
-    if message:
-        yield Validation.warning(
-            "ROS.19",
-            message,
-            modelObject=modelObjects,
-            principal_currency_value=principal_currency_value
-        )
+                yield Validation.warning(
```

Comment:
Can this warning potentially fire multiple times within the containing loop now?